plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,468 @@
+## append the corresponding backend or agent API path (e.g. /events or 
+## /api/core/v2/namespaces/:entity_namespace/events/:entity_name/:check_name).
+## 
+## NOTE: if backend_api_url and agent_api_url and api_key are set, the output 
+## plugin will use backend_api_url. If backend_api_url and agent_api_url are 
+## not provided, the output plugin will default to use an agent_api_url of 
+## http://127.0.0.1:3031
+## 
+# backend_api_url = "http://127.0.0.1:8080"
+# agent_api_url = "http://127.0.0.1:3031"
+
+## API KEY is the Sensu Backend API token 
+## Generate a new API token via: 
+## 
+## $ sensuctl cluster-role create telegraf --verb create --resource events,entities
+## $ sensuctl cluster-role-binding create telegraf --cluster-role telegraf --group telegraf
+## $ sensuctl user create telegraf --group telegraf --password REDACTED 
+## $ sensuctl api-key grant telegraf
+##
+## For more information on Sensu RBAC profiles & API tokens, please visit: 
+## - https://docs.sensu.io/sensu-go/latest/reference/rbac/
+## - https://docs.sensu.io/sensu-go/latest/reference/apikeys/ 
+## 
+# api_key = "${SENSU_API_KEY}"
+
+## Optional TLS Config
+# tls_ca = "/etc/telegraf/ca.pem"
+# tls_cert = "/etc/telegraf/cert.pem"
+# tls_key = "/etc/telegraf/key.pem"
+## Use TLS but skip chain & host verification
+# insecure_skip_verify = false
+
+## Timeout for HTTP message
+# timeout = "5s"
+
+## HTTP Content-Encoding for write request body, can be set to "gzip" to
+## compress body or "identity" to apply no encoding.
+# content_encoding = "identity"
+
+## Sensu Event details 
+## 
+## NOTE: if the output plugin is configured to send events to a 
+## backend_api_url and entity_name is not set, the value returned by 
+## os.Hostname() will be used; if the output plugin is configured to send
+## events to an agent_api_url, entity_name and entity_namespace are not used. 
+# [outputs.sensu-go.entity]
+#   name = "server-01"
+#   namespace = "default"
+
+# [outputs.sensu-go.tags]
+#   source = "telegraf"
+
+# [outputs.sensu-go.metrics]
+#   handlers = ["elasticsearch","timescaledb"]
+`
+
+// Description provides a description of the plugin
+func (s *Sensu) Description() string {
+	return "Send aggregate metrics to Sensu Monitor"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (s *Sensu) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Sensu) SetSerializer(serializer serializers.Serializer) {
+	s.serializer = serializer
+}
+
+func (s *Sensu) createClient(ctx context.Context) (*http.Client, error) {
+	tlsCfg, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: s.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (s *Sensu) Connect() error {
+	if s.Timeout.Duration == 0 {
+		s.Timeout.Duration = defaultClientTimeout
+	}
+
```

Comment:
You should also check if the other options are valid to avoid user configuration errors. E.g. currently only `gzip` and `identity` is supported for `ContentEncoding`. You maybe should error out if the user specifies something else. Check the `internal.choice` package providing some convenient functions for doing so.

Reply:
We're doing config validation now.